test/LockedNORI.t.sol

```diff
@@ -195,22 +215,75 @@ contract LockedNORITest is Test, ERC777ERC1820 {
       address(this)
     );
 
-    bpNori = new TestToken777();
-    assertEq(bpNori.balanceOf(address(this)), SEED_AMOUNT);
+    erc777 = deployMockERC777();
+    assertEq(erc777.balanceOf(address(this)), SEED_AMOUNT);
+
+    erc20 = deployMockERC20();
+    assertEq(erc20.balanceOf(address(this)), SEED_AMOUNT);
+
+    lNori = deployLockedNORIV2(address(erc20));
+    helper = new LockedNORIV2Helper();
+    signer = new PermitSigner();
+  }
+
+  event Approval(address indexed owner, address indexed spender, uint256 value);
+
+  function testBatchGrantCreation() public {
+    address recipient = vm.addr(0xa11ce);
+    uint256 grantAdminKey = 0xabcdef;
+    address grantAdmin = vm.addr(grantAdminKey);
+    erc20.transfer(grantAdmin, GRANT_AMOUNT);
+    assertEq(erc20.balanceOf(address(grantAdmin)), GRANT_AMOUNT);
+    lNori.grantRole(lNori.TOKEN_GRANTER_ROLE(), grantAdmin);
+
+    uint256 deadline = 1 days;
+
+    bytes memory params = helper.getSimpleGrantCreationParamsEncoded(
+      recipient,
+      1 days
+    );
+    vm.startPrank(grantAdmin);
+    bytes32 digest = signer.digestPermitCall(
+      address(erc20),
+      address(lNori),
+      GRANT_AMOUNT,
+      deadline
+    );
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(grantAdminKey, digest);
+
+    uint256[] memory amounts = new uint256[](1);
+    amounts[0] = GRANT_AMOUNT;
+    bytes[] memory data = new bytes[](1);
+    data[0] = params;
 
-    lNori = deployLockedNORIV2();
+    vm.expectEmit(true, true, true, false);
+    emit Approval(address(grantAdmin), address(lNori), GRANT_AMOUNT);
+    lNori.batchCreateGrants(amounts, data, deadline, v, r, s);
+
+    helper.assertSimplePastGrant(address(lNori), lNori.getGrant(recipient));
+    assertEq(erc20.balanceOf(grantAdmin), 0);
+    assertEq(erc20.balanceOf(address(lNori)), GRANT_AMOUNT);
+
+    assertEq(lNori.balanceOf(recipient), GRANT_AMOUNT);
+    assertEq(lNori.vestedBalanceOf(recipient), 0);
+    assertEq(lNori.unlockedBalanceOf(recipient), 0);
+
+    vm.warp(3656 days);
```

Comment:
is this a typo or is this just accounting for leap years?

Reply:
typo, thanks!